nselib/http.lua

```diff
@@ -1571,7 +1571,9 @@ function parse_redirect(host, port, path, response)
   if ( not(u.host) ) then
     -- we're dealing with a relative url
     u.host = stdnse.get_hostname(host)
-    u.path = ((u.path:sub(1,1) == "/" and "" ) or "/" ) .. u.path -- ensuring leading slash
+    if ( u.path:sub(1,1) ~= "/") then
```

Comment:
Yes, it seems url.absolute() does the job perfectly. It simplified the code a lot. After addressing nnposter's comments, feel free to commit this patch as this is a bug affecting the redirects independent to cookie management feature you are working on.